src/java/com/wolfssl/provider/jsse/WolfSSLEngineHelper.java

```diff
@@ -175,7 +175,7 @@ protected void setProtocols(String[] p) throws IllegalArgumentException {
 
     /* gets all supported protocols */
     protected String[] getAllProtocols() {
-        return WolfSSL.getProtocols();
+        return WolfSSL.getProtocolsMask(ssl.getOptions());
```

Comment:
We should consider adding a check on "ssl" for null before calling "getOptions()" on it to prevent a possible NullPointerException.

Maybe we can fall back to "WolfSSL.getProtocols()" if "ssl" is null?

Reply:
fixed